### PR TITLE
chore: add prefixes to log messages

### PR DESCRIPTION
**Pull request by knqyf263**
## Description
Adding prefixes for each artifact.

### Before

```
2024-10-01T15:43:48+04:00       INFO    [db] Need to update DB
2024-10-01T15:43:48+04:00       INFO    Downloading vulnerability DB...
2024-10-01T15:43:48+04:00       INFO    Downloading  artifact...        repo="ghcr.io/aquasecurity/trivy-db:2"
53.85 MiB / 53.85 MiB [---------------------------------------------------------------------------------------------------------------] 100.00% 23.26 MiB p/s 2.5s
2024-10-01T15:43:51+04:00       INFO    Artifact successfully downloaded        repo="ghcr.io/aquasecurity/trivy-db:2"
```

### After

```
2024-10-01T18:16:51+04:00       INFO    [vulndb] Need to update DB
2024-10-01T18:16:51+04:00       INFO    [vulndb] Downloading vulnerability DB...
2024-10-01T18:16:51+04:00       INFO    [vulndb] Downloading artifact...        repo="ghcr.io/aquasecurity/trivy-db:2"
53.87 MiB / 53.87 MiB [---------------------------------------------------------------------------------------------------------------] 100.00% 28.86 MiB p/s 2.1s
2024-10-01T18:16:55+04:00       INFO    [vulndb] Artifact successfully downloaded       repo="ghcr.io/aquasecurity/trivy-db:2"
```

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [x] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/db/db.go [hunk 1]
```diff
@@ -93,30 +93,30 @@ func NewClient(dbDir string, quiet bool, opts ...Option) *Client {
...
-			log.Error("The first run cannot skip downloading DB")
+			log.ErrorContext(ctx, "The first run cannot skip downloading DB")
 			return false, xerrors.New("--skip-update cannot be specified on the first run")
 		}
 		meta = metadata.Metadata{Version: db.SchemaVersion}
 	}
 
 	if db.SchemaVersion < meta.Version {
-		log.Error("The Trivy version is old. Update to the latest version.", log.String("version", cliVersion))
+		log.ErrorContext(ctx, "The Trivy version is old. Update to the latest version.", log.String("version", cliVersion))
```
Comment (simar7) [inline pkg/db/db.go:105]:
```suggestion
		log.ErrorContext(ctx, "Trivy version is old. Update to the latest version.", log.String("version", cliVersion))
```

File: pkg/javadb/client.go [hunk 2]
```diff
@@ -73,33 +74,36 @@ func (u *Updater) Update() error {
 		if err = metac.Update(meta); err != nil {
 			return xerrors.Errorf("Java DB metadata update error: %w", err)
 		}
-		log.Info("The Java DB is cached for 3 days. If you want to update the database more frequently, " +
+		log.InfoContext(ctx, "The Java DB is cached for 3 days. If you want to update the database more frequently, "+
```
Comment (simar7) [inline pkg/javadb/client.go:77]:
```suggestion
		log.InfoContext(ctx, "Java DB is cached for 3 days. If you want to update the database more frequently, "+
```

**Comment by nikpivkin** [comment]
[Here](https://github.com/aquasecurity/trivy/blob/main/pkg/policy/policy.go#L214-L228) it is logged without the prefix

File: pkg/log/logger.go [hunk 3]
```diff
@@ -23,6 +23,9 @@ const (
 	PrefixMisconfiguration = "misconfig"
 	PrefixSecret           = "secret"
 	PrefixLicense          = "license"
+	PrefixVulnerabilityDB  = "vulndb"
+	PrefixJavaDB           = "javadb"
+	PrefixChecksBundle     = "checks"
```
Comment (DmitriyLewen) [inline pkg/log/logger.go:28]: `checks` looks too generic.
Perhaps we want to use `checks-bundle` for this?

**Review by DmitriyLewen — COMMENTED** [review]
LGTM

Left one comment

**Comment by knqyf263** [comment]
> [Here](https://github.com/aquasecurity/trivy/blob/main/pkg/policy/policy.go#L214-L228) it is logged without the prefix

@nikpivkin It is called via `NewVersionInfo`, which doesn't have a context as the argument. That's why I didn't change it, but I eventually used `context.TODO()` so we can fix it later.
https://github.com/aquasecurity/trivy/pull/7625/commits/6d648a8f19cf8b2ad2d69d1dc63d5eacb16f413d

File: pkg/log/logger.go (hunk 3 above)
Comment (knqyf263) [reply pkg/log/logger.go]: Good point, I was actually thinking about it and didn't have a good idea. Then, I used `checks`. On second thoughts, I chose `misconfig`.
https://github.com/aquasecurity/trivy/pull/7625/commits/d8bfa99829c98458d12385f9215efb52eb0f688a

**Comment by nikpivkin** [comment]
@knqyf263 Is there an advantage of using context for logging over using a logger with a prefix as a field? Then there would be no need for context.

**Review by nikpivkin — APPROVED** [review]
LGTM

**Review by afdesk — APPROVED** [review]

**Comment by knqyf263** [comment]
> @knqyf263 Is there an advantage of using context for logging over using a logger with a prefix as a field? Then there would be no need for context.

It seems more intuitive and preferable to define a new logger rather than a context if the prefix is closed within a specific package. In this case, however, the prefix needed to be propagated. 

Other approaches are:

1. Re-define a new logger in each package

```
diff --git a/pkg/commands/artifact/run.go b/pkg/commands/artifact/run.go
index eeca48436..d85f54226 100644
--- a/pkg/commands/artifact/run.go
+++ b/pkg/commands/artifact/run.go
@@ -618,8 +618,8 @@ func (r *runner) scan(ctx context.Context, opts flag.Options, initializeScanner
 }

 func initMisconfScannerOption(ctx context.Context, opts flag.Options) (misconf.ScannerOption, error) {
-       ctx = log.WithContextPrefix(ctx, log.PrefixMisconfiguration)
-       log.InfoContext(ctx, "Misconfiguration scanning is enabled")
+       logger := log.WithPrefix(log.PrefixMisconfiguration)
+       logger.Info("Misconfiguration scanning is enabled")

        var downloadedPolicyPaths []string
        var disableEmbedded bool
@@ -630,7 +630,7 @@ func initMisconfScannerOption(ctx context.Context, opts flag.Options) (misconf.S
                        log.ErrorContext(ctx, "Falling back to embedded checks", log.Err(err))
                }
        } else {
-               log.DebugContext(ctx, "Checks successfully loaded from disk")
+               logger.Debug("Checks successfully loaded from disk")
                disableEmbedded = true
        }

diff --git a/pkg/commands/operation/operation.go b/pkg/commands/operation/operation.go
index ac52eee7f..e4f0feba2 100644
--- a/pkg/commands/operation/operation.go
+++ b/pkg/commands/operation/operation.go
@@ -82,6 +82,7 @@ func InitBuiltinChecks(ctx context.Context, cacheDir string, quiet, skipUpdate b
        mu.Lock()
        defer mu.Unlock()

+       logger := log.WithPrefix(log.PrefixMisconfiguration)
        client, err := policy.NewClient(cacheDir, quiet, checkBundleRepository)
        if err != nil {
                return nil, xerrors.Errorf("check client error: %w", err)
@@ -96,8 +97,8 @@ func InitBuiltinChecks(ctx context.Context, cacheDir string, quiet, skipUpdate b
        }

        if needsUpdate {
-               log.InfoContext(ctx, "Need to update the built-in checks")
-               log.InfoContext(ctx, "Downloading the built-in checks...")
+               logger.Info("Need to update the built-in checks")
+               logger.Info("Downloading the built-in checks...")
                if err = client.DownloadBuiltinChecks(ctx, registryOpts); err != nil {
                        return nil, xerrors.Errorf("failed to download built-in policies: %w", err)
                }
```

2. Pass a logger

```
+func InitBuiltinChecks(ctx context.Context, logger *log.Logger, cacheDir string, quiet, skipUpdate bool, checkBundleRepository string, registryOpts ftypes.RegistryOptions) ([]string, error) {
        mu.Lock()
        defer mu.Unlock()

@@ -96,8 +96,8 @@ func InitBuiltinChecks(ctx context.Context, cacheDir string, quiet, skipUpdate b
        }

        if needsUpdate {
-               log.InfoContext(ctx, "Need to update the built-in checks")
-               log.InfoContext(ctx, "Downloading the built-in checks...")
+               logger.Info("Need to update the built-in checks")
+               logger.Info("Downloading the built-in checks...")
```

Using a context looks better. Does it answer your question?

**Comment by nikpivkin** [comment]
@knqyf263 Yeah, thanks.

**Comment by knqyf263** [comment]
I believe that `context.Context` should be used as little as possible because it is difficult to see what values are in that, but I thought that `context.Context` might be suitable (in terms of its name) when a common "context" (in this case a logging prefix) needs to be shared across multiple packages. However, I'm always open to other ideas!

**Pull request merged**